atomic_reactor/inner.py

```diff
@@ -326,9 +326,9 @@ class ImageBuildWorkflowData(ISerializer):
     # "path/to/file" -> "content"
     files: Dict[str, str] = field(default_factory=dict)
 
-    # List of RPMs that go into the final result, as per utils.rpm.parse_rpm_output
+    # Per platform List of RPMs that go into the final result
     # Each RPM inside is a mapping containing the name, version, release and other attributes.
-    image_components: Optional[List[RpmComponent]] = None
+    image_components: Dict[str, List[RpmComponent]] = field(default_factory=dict)
```

Comment:
Was there ever a difference between `image_components == None` and `image_components == []`? If yes, we should still make this `Optional`

Reply:
There is a difference but it's going to be set to either the result of `rpmqa` or `flatpak_create_oci` and in both cases, it'll have keys set to platform so it's not going to be `None`

Reply:
Oh, true, after some plugin sets this it cannot "falsy" dict since it has platform keys. Should be OK then :+1: